### PR TITLE
[somfytahoma] Added support somfy curtains

### PR DESCRIPTION
**Pull request by TobiasAmmann**
#### Description
This pull request adds all needed classes to support curtains in the Somfy Tahoma binding. Documentation is updated with the thing

#### Testing
The extension is tested with the Somfy Irismo 35 curtain rails.
Static code analysis didn't show any additional issues

### CODE REVIEW

**Comment by J-N-K** [comment]
rebuild label not working. triggering it by closing&open

**Comment by octa22** [comment]
@TobiasAmmann  Could you please add THING_TYPE_CURTAIN to the getSupportedThingTypes() method? (discovery service)
Thanks
Ondrej

File: bundles/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/internal/handler/SomfyTahomaCurtainHandler.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+                sendCommand(cmd, param);
+            }
+        }
+    }
+
+    private String getTahomaCommand(String command) {
+        switch (command) {
+            case "OFF":
+            case "DOWN":
+            case "CLOSE":   return COMMAND_CLOSE;
```
Comment (cweitkamp) [inline bundles/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/internal/handler/SomfyTahomaCurtainHandler.java]: Please apply the formatter.

File: bundles/org.openhab.binding.somfytahoma/README.md [hunk 2]
```diff
@@ -58,7 +59,7 @@ Please see the example below.
 | gateway       | status  | status of your Tahoma gateway |
 | gate          | gate_command | used for controlling your gate (open, close, stop, pedestrian) |
 | gate          | gate_state | get state of your gate |
-| roller shutter, screen, venetian blind, garage door, awning, window, pergola | control |  device controller which reacts to commands UP/DOWN/STOP + closure 0-100 |
+| roller shutter, screen, venetian blind, garage door, awning, window, pergola, curtain | control |  device controller which reacts to commands UP/DOWN/STOP + closure 0-100 |
```
Comment (cweitkamp) [inline bundles/org.openhab.binding.somfytahoma/README.md]: Only nitpicking: Can you apply the markdown table formatter (see https://www.tablesgenerator.com/markdown_tables)?

**Comment by kaikreuzer** [comment]
> @kaikreuzer I cannot remember if I asked it before: Can we add @octa22 to the contributors team? 

Sure thing - done!

File: bundles/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/internal/handler/SomfyTahomaCurtainHandler.java [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+        super(thing);
+        stateNames = new HashMap<String, String>() {{
+            put(CONTROL, "core:ClosureState");
+        }};
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        logger.debug("Received command {} for channel {}", command, channelUID);
+        if (!CONTROL.equals(channelUID.getId())) {
```
Comment (octa22) [inline bundles/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/internal/handler/SomfyTahomaCurtainHandler.java]: What about to 
1) make the method ```getTahomaCommand``` in the ```SomfyTahomaRollerShutterHandler``` class protected 
2) inherit ```SomfyTahomaCurtainHandler``` from ```SomfyTahomaRollerShutterHandler``` and remove ```handleCommand``` method
3) add ```@override``` to ```getTahomaCommand``` in ```SomfyTahomaCurtainHandler```

**Comment by J-N-K** [comment]
@TobiasAmmann Did you see @octa22's comment?

**Comment by Hilbrand** [comment]
Closing because replaced by PR #6161

**Pull request closed without merging**